deploy/template.yaml

```diff
@@ -1203,22 +1255,31 @@ Resources:
             Action:
               - kms:*
             Resource: "*"
+
+  S3KmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: 2012-10-17
+        Statement:
           - Effect: Allow
             Principal:
-              Service: "dynamodb.amazonaws.com"
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - "kms:Encrypt*"
+              - "kms:Decrypt*"
+            Resource: "*"
+          - Effect: Allow
+            Principal:
+              AWS: !FindInMap
+                - EnvironmentConfiguration
+                - !Ref AWS::AccountId
+                - spotAccountArn
             Action:
               - "kms:Encrypt*"
               - "kms:Decrypt*"
-              - "kms:ReEncrypt*"
-              - "kms:GenerateDataKey*"
-              - "kms:Describe*"
             Resource: "*"
```

Comment:
Might be worth adding a `kms:ViaService` condition of `s3.amazonaws.com` here ... this will ensure we can _only_ use the key for encrypting/decryption from S3.

Reply:
Condition to restrict usage to s3 crypto has been added.

Reply:
Minor point, you don't need the `*` in `kms:Decrypt*` ... it doesn't really matter because, at the moment, there is only one KMS API action that begins with `Decrypt` and that is `Decrypt` itself! However, if AWS added a new method called `kms:DecryptAllTheSecrets` you would be inadvertently be giving access to that method by including the `*` here. I think its best to avoid the `*` and be explicit.